### PR TITLE
Fix #1434 2.55.1 Throws when using where query against "JSON Object or ANY" Type

### LINKED ISSUE #1434: TypeError: Cannot read property 'properties' of undefined using @2.55.1
# Description/Steps to reproduce
Running loopback in docker node-apline 6.11

Working fine with 2.54.1. And throws following error when using 2.55.1.

# Expected result
Should be working without throwing error
```
TypeError: Cannot read property 'properties' of undefined
    at Function.DataAccessObject._coerce (/usr/src/app/node_modules/loopback-datasource-juggler/lib/dao.js:1615:60)
    at Function.DataAccessObject._normalize (/usr/src/app/node_modules/loopback-datasource-juggler/lib/dao.js:1511:8)
    at Function.find (/usr/src/app/node_modules/loopback-datasource-juggler/lib/dao.js:1855:10)
    at prepareForTokenInvalidation (/usr/src/app/node_modules/loopback/common/models/user.js:889:15)
    at notifySingleObserver (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:104:22)
    at iterate (/usr/src/app/node_modules/loopback-datasource-juggler/node_modules/async/lib/async.js:181:13)
    at Object.async.eachSeries (/usr/src/app/node_modules/loopback-datasource-juggler/node_modules/async/lib/async.js:197:9)
    at doNotify (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:101:11)
    at doNotify (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:99:49)
    at doNotify (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:99:49)
    at doNotify (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:99:49)
    at Function.ObserverMixin._notifyBaseObservers (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:122:5)
    at Function.ObserverMixin.notifyObserversOf (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:97:8)
    at Function.ObserverMixin._notifyBaseObservers (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:120:15)
    at Function.ObserverMixin.notifyObserversOf (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:97:8)
    at Function.ObserverMixin._notifyBaseObservers (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:120:15)
    at Function.ObserverMixin.notifyObserversOf (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:97:8)
    at Function.ObserverMixin._notifyBaseObservers (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:120:15)
    at Function.ObserverMixin.notifyObserversOf (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:97:8)
    at Function.ObserverMixin._notifyBaseObservers (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:120:15)
    at Function.ObserverMixin.notifyObserversOf (/usr/src/app/node_modules/loopback-datasource-juggler/lib/observer.js:97:8)
    at /usr/src/app/node_modules/loopback-datasource-juggler/lib/dao.js:2608:11
```

# Additional information

### PR DESCRIPTION
**Pull request by yshing**
#### Related issues

- #1434 

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: lib/dao.js [hunk 1]
```diff
@@ -1610,9 +1610,12 @@ DataAccessObject._coerce = function(where, props) {
       if (props[model]) {
         var clause = {};
         clause[prop] = where[p];
-        where[p] = Array.isArray(props[model].type) ?
-          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
-          self._coerce(clause, props[model].type.definition.properties)[prop];
+        var type = Array.isArray(props[model].type) ? props[model].type[0] : props[model].type;
+        if (type && type.definition && type.definition.properties) {
+          where[p] = self._coerce(clause, type.definition.properties)[prop];
+          continue;
```
Comment (kjdelisle) [inline lib/dao.js]: Minor nitpick: this `continue` call is unnecessary, as it will fall through to the next one.

**Comment by kjdelisle** [comment]
@slnode test please

**Comment by kjdelisle** [comment]
@yshing Thank you for your contribution! :tada:

**Pull request merged**